### PR TITLE
feat(shared): add WORKSPACE_COUNT_IN_PROJECT in LimitMetric

### PR DESCRIPTION
**Pull request by CHLK**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
add WORKSPACE_COUNT_IN_PROJECT in LimitMetric;
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties [hunk 1]
```diff
@@ -142,6 +142,7 @@ com.oceanbase.odc.LimitMetric.TABLE_NAME_LENGTH=table name length
 com.oceanbase.odc.LimitMetric.WORKSHEET_CHANGE_COUNT=number of changed worksheets
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/BusinessMessages.properties:142]: The number of changed worksheets

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties [hunk 2]
```diff
@@ -142,6 +142,7 @@ com.oceanbase.odc.LimitMetric.TABLE_NAME_LENGTH=table name length
...
 com.oceanbase.odc.LimitMetric.WORKSHEET_SAME_LEVEL_COUNT=number of same level worksheets
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/BusinessMessages.properties:143]: The number of worksheets at the same level

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties [hunk 3]
```diff
@@ -142,6 +142,7 @@ com.oceanbase.odc.LimitMetric.TABLE_NAME_LENGTH=table name length
...
 com.oceanbase.odc.LimitMetric.WORKSHEET_COUNT_IN_PROJECT=number of project worksheets
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/BusinessMessages.properties:144]: The number of worksheets in the project

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties [hunk 4]
```diff
@@ -142,6 +142,7 @@ com.oceanbase.odc.LimitMetric.TABLE_NAME_LENGTH=table name length
...
+com.oceanbase.odc.LimitMetric.WORKSPACE_COUNT_IN_PROJECT=number of project workspaces
```
Comment (JessieWuJiexi) [inline server/odc-core/src/main/resources/i18n/BusinessMessages.properties:145]: The number of workspaces in the project

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties (hunk 1 above)
Comment (CHLK) [reply server/odc-core/src/main/resources/i18n/BusinessMessages.properties:142]: okay, i'll make this change.

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties (hunk 2 above)
Comment (CHLK) [reply server/odc-core/src/main/resources/i18n/BusinessMessages.properties:143]: okay, i'll make this change.

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties (hunk 3 above)
Comment (CHLK) [reply server/odc-core/src/main/resources/i18n/BusinessMessages.properties:144]: okay, i'll make this change.

File: server/odc-core/src/main/resources/i18n/BusinessMessages.properties (hunk 4 above)
Comment (CHLK) [reply server/odc-core/src/main/resources/i18n/BusinessMessages.properties:145]: okay, i'll make this change.

**Pull request closed without merging**